tests/test_core_circuit_breaker.py

```
@@ -0,0 +1,352 @@
+        assert metrics.successes_total == 0
+        assert metrics.trips_total == 0
+
+    def test_expected_exceptions_only(self, breaker):
+        """Test that only expected exceptions count as failures"""
+        # Custom exception that's not expected
+        class CustomError(Exception):
+            pass
+
+        breaker.expected_exceptions = (ValueError,)
+
+        mock_func_success = AsyncMock(return_value="success")
```

Ruff (0.14.10)

Local variable `mock_func_success` is assigned to but never used

Remove assignment to unused variable `mock_func_success`

(F841)